plugins/filter_aws/aws.c

```diff
@@ -377,26 +489,18 @@ static int get_ec2_metadata(struct flb_filter_aws *ctx)
 {
     int ret;
 
-    if (ctx->use_v2 == FLB_TRUE && !ctx->imds_v2_token) {
-        ret = get_ec2_token(ctx);
-
-        if (ret < 0) {
-            return -1;
-        }
-    }
-
     if (ctx->instance_id_include && !ctx->instance_id) {
-        ret = get_metadata(ctx, FLB_FILTER_AWS_IMDS_INSTANCE_ID_PATH,
-                           &ctx->instance_id, &ctx->instance_id_len);
-
+        ret = flb_aws_imds_request(ctx->client_imds, FLB_AWS_IMDS_INSTANCE_ID_PATH,
+                                   &ctx->instance_id,
+                                   &ctx->instance_id_len);
         if (ret < 0) {
```

Comment:
For each of these, can you Please add a clear error message telling the user which metadata request failed like: "Failed to get instance ID" for this one.